python_modules/libraries/dagster-airlift/dagster_airlift/core/monitoring_job/event_stream.py

```diff
@@ -0,0 +1,295 @@
+from collections.abc import Iterator
+from itertools import chain
+from typing import Optional
+
+from dagster import AssetMaterialization
+from dagster._core.events import (
+    AssetMaterializationPlannedData,
+    DagsterEvent,
+    JobFailureData,
+    StepMaterializationData,
+)
+from dagster._core.execution.context.op_execution_context import OpExecutionContext
+from dagster._core.remote_representation.origin import RemoteJobOrigin
+from dagster._core.storage.dagster_run import DagsterRun, DagsterRunStatus
+from dagster._core.storage.tags import EXTERNAL_JOB_SOURCE_TAG_KEY
+from dagster._core.utils import make_new_run_id
+from dagster._record import record
+from dagster._time import datetime_from_timestamp
+from dagster_airlift.constants import DAG_ID_TAG_KEY, DAG_RUN_ID_TAG_KEY
+from dagster_airlift.core.airflow_defs_data import AirflowDefinitionsData
+from dagster_airlift.core.airflow_instance import AirflowInstance
+from dagster_airlift.core.monitoring_job.utils import (
+    get_dagster_run_for_airflow_repr,
+    structured_log,
+)
+from dagster_airlift.core.runtime_representations import DagRun, TaskInstance
+
+
+@record
+class DagRunStarted:
```

Comment:
should these things all share a base class?

Reply:
eh